debian/patches/ocf-suexec.patch

```diff
@@ -1,23 +1,26 @@
 --- a/support/suexec-ocf.c
 +++ b/support/suexec-ocf.c
-@@ -14,6 +14,7 @@
-  * limitations under the License.
-  */
- 
-+
- /*
-  * suexec.c -- "Wrapper" support program for suEXEC behaviour for Apache
-  *
-@@ -69,6 +70,8 @@
+@@ -69,6 +69,9 @@
  
  #define AP_ENVBUF 256
  
 +#define AP_DOC_ROOT "/services"
++#define AP_SUEXEC_ROOT "/services/http/suexec"
 +
  extern char **environ;
  static FILE *log = NULL;
  
-@@ -579,11 +582,15 @@
+@@ -532,7 +535,8 @@
+
+     if (strlen(cwd) > strlen(dwd))
+         strncat(dwd, "/", 1);
+-    if ((strncmp(cwd, dwd, strlen(dwd))) != 0) {
++    if ((strncmp(cwd, dwd, strlen(dwd))) != 0 &&
++        (strncmp(cwd, AP_SUEXEC_ROOT, strlen(AP_SUEXEC_ROOT)) != 0)) {
```

Comment:
since `AP_SUEXEC_ROOT` is a `#define`, you could use `sizeof(AP_SUEXEC_ROOT)` (`- 1` to be totally equivalent but i don't think it matters in this case) instead to avoid the runtime string length calculation

the compiler probably optimizes to that anyway?

/shrug

Reply:
I verified that the compiler just optimizes this call out since it's a constant (with `gcc` anyway). This just becomes a `mov  $0x15, %esi`, where `0x15` is a constant for 21, the length of the string. Given this is optimized out, I'd rather keep it like this, since I think it's clearer than doing `sizeof(AP_SUEXEC_ROOT)` and subtracting 1, or just ignoring the difference of one.

Reply:
that's fine. i think the only advantage is that `sizeof` is "obviously" done at compile-time whereas `strlen` makes it less clear whether it's runtime or if the compiler will optimize it out, but the performance doesn't matter here anyway